inception-search-mtas/src/test/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndexTest.java

```diff
@@ -0,0 +1,557 @@
+import de.tudarmstadt.ukp.clarin.webanno.api.dao.initializers.PartOfSpeechLayerInitializer;
+import de.tudarmstadt.ukp.clarin.webanno.api.dao.initializers.TokenLayerInitializer;
+import de.tudarmstadt.ukp.clarin.webanno.curation.storage.CurationDocumentService;
+import de.tudarmstadt.ukp.clarin.webanno.curation.storage.CurationDocumentServiceImpl;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationDocument;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.clarin.webanno.model.SourceDocument;
+import de.tudarmstadt.ukp.clarin.webanno.project.ProjectServiceImpl;
+import de.tudarmstadt.ukp.clarin.webanno.security.UserDao;
+import de.tudarmstadt.ukp.clarin.webanno.security.UserDaoImpl;
+import de.tudarmstadt.ukp.clarin.webanno.security.model.Role;
+import de.tudarmstadt.ukp.clarin.webanno.security.model.User;
+import de.tudarmstadt.ukp.clarin.webanno.support.ApplicationContextProvider;
+import de.tudarmstadt.ukp.dkpro.core.io.text.TextReader;
+import de.tudarmstadt.ukp.dkpro.core.io.text.TextWriter;
+import de.tudarmstadt.ukp.inception.kb.KnowledgeBaseService;
+import de.tudarmstadt.ukp.inception.kb.KnowledgeBaseServiceImpl;
+import de.tudarmstadt.ukp.inception.search.FeatureIndexingSupport;
+import de.tudarmstadt.ukp.inception.search.FeatureIndexingSupportRegistry;
+import de.tudarmstadt.ukp.inception.search.FeatureIndexingSupportRegistryImpl;
+import de.tudarmstadt.ukp.inception.search.PrimitiveUimaIndexingSupport;
+import de.tudarmstadt.ukp.inception.search.SearchResult;
+import de.tudarmstadt.ukp.inception.search.SearchService;
+import de.tudarmstadt.ukp.inception.search.SearchServiceImpl;
+import de.tudarmstadt.ukp.inception.search.index.PhysicalIndexFactory;
+import de.tudarmstadt.ukp.inception.search.index.PhysicalIndexRegistry;
+import de.tudarmstadt.ukp.inception.search.index.PhysicalIndexRegistryImpl;
+import de.tudarmstadt.ukp.inception.search.scheduling.IndexScheduler;
+import net.jodah.concurrentunit.Waiter;
+
+/**
+ * The Class MtasDocumentIndexTest.
+ */
+
+@RunWith(SpringRunner.class)
+@EnableAutoConfiguration
+@SpringBootTest(webEnvironment = WebEnvironment.MOCK)
+@EntityScan({ "de.tudarmstadt.ukp.clarin.webanno.model",
+        "de.tudarmstadt.ukp.inception.search.model",
+        "de.tudarmstadt.ukp.clarin.webanno.security.model" })
+@TestPropertySource(locations = "classpath:MtasDocumentIndexTest.properties")
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@DataJpaTest
+@Transactional
+public class MtasDocumentIndexTest
+{
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    // Number of miliseconds to wait for the indexing to finish. This time must be enough
+    // to allow the index be built before the query is made. Otherwise, it could affect the
+    // test results. If this happens, a largest value could allow the test to pass.
+    private final int WAIT_TIME = 3000;
+
+    private final String temporaryFolderPath = "target/MtasDocumentIndexTest";
```

Comment:
You can use 

```
  @Rule
  public TemporaryFolder folder= new TemporaryFolder();
```

instead of managing a temporary folder by yourself.

Reply:
I usually prefer using `DkproTestContext` (which is also a rule) and `DkproTestContext.getTestOutputFolder()` because that places the test output folder under `test-output` in the `target` folder where the data can actually be inspected after the tests. The `TemporaryFolder` is removed after the tests are complete which I find terribly annoying when writing/debugging tests.

Reply:
I haven't been able to use a temporary folder rule, since I need the temporary folder to be initialized in the text context. Cf. https://stackoverflow.com/questions/24922153/spring-config-with-junit-temporary-folder